data/data-pipeline/data_pipeline/etl/downloader.py

```diff
@@ -0,0 +1,96 @@
+import os
+import uuid
+import urllib3
+import requests
+import zipfile
+
+from pathlib import Path
+from data_pipeline.config import settings
+
+
+class Downloader:
+    """A simple class to encapsulate the download capabilities of the application"""
+
+    @classmethod
+    def download_file_from_url(
+        cls,
+        file_url: str,
+        download_file_name: Path,
+        verify: bool = True,
+    ) -> str:
+        """Downloads a file from a remote URL location and returns the file location.
+
+        Args:
+                file_url (str): URL where the zip file is located
+                download_file_name (pathlib.Path): file path where the file will be downloaded (called downloaded.zip by default)
+                verify (bool): A flag to check if the certificate is valid. If truthy, an invalid certificate will throw an
+                error (optional, default to False)
+
+        Returns:
+                None
+
+        """
+        # disable https warning
+        urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+        download_file_name.parent.mkdir(parents=True, exist_ok=True)
+
+        response = requests.get(
+            file_url, verify=verify, timeout=settings.REQUESTS_DEFAULT_TIMOUT
+        )
+        if response.status_code == 200:
+            file_contents = response.content
+        else:
+            raise Exception(
+                f"HTTP response {response.status_code} from url {file_url}. Info: {response.content}"
+            )
+
+        # Write the contents to disk.
+        file = open(download_file_name, "wb")
+        file.write(file_contents)
+        file.close()
+
+        return download_file_name
+
+    @classmethod
+    def download_zip_file_from_url(
+        cls,
+        file_url: str,
+        unzipped_file_path: Path,
+        verify: bool = True,
+    ) -> None:
+        """Downloads a zip file from a remote URL location and unzips it in a specific directory, removing the temporary file after
+
+        Args:
+                file_url (str): URL where the zip file is located
+                unzipped_file_path (pathlib.Path): directory and name of the extracted file
+                verify (bool): A flag to check if the certificate is valid. If truthy, an invalid certificate will throw an
+                error (optional, default to False)
+
+        Returns:
+                None
+
+        """
+        # file_id allows us to evade race conditions on parallel ETLs
+        file_id = uuid.uuid4()
+        dir_id = uuid.uuid4()
+
+        zip_download_path = (
+            settings.DATA_PATH
+            / "tmp"
+            / "downloads"
+            / f"{dir_id}"
```

Comment:
double-uuid! no one is finding that file. For real tho, do you actually need double UUIDs?  like the file could always be called `download.zip`

Reply:
Prooooobably. 😄 I will say the point is for no one to find the file. This is a temporary path for downloading, and the file is moved to the actual destination later. The first UUID is fixing the fact that previously you had to pass in the temp directory, and I thought "nobody cares what the temp directory is, they just care that the file makes it to the final destination." The second UUID is inherited and can just be removed. So good catch.